dpnp/backend/extensions/blas/gemm_batch.cpp

```diff
@@ -155,12 +155,12 @@ std::pair<sycl::event, sycl::event>
                size_t stridec,
                const std::vector<sycl::event> &depends = {})
 {
-    if (!dpctl::utils::queues_are_compatible(
-            exec_q,
-            {matrixA.get_queue(), matrixB.get_queue(), resultC.get_queue()}))
-    {
-        throw py::value_error(
-            "USM allocations are not compatible with the execution queue.");
+    const int matrixA_nd = matrixA.get_ndim();
```

Comment:
Do we need to have checks `dpctl::tensor::validation::CheckWritable::throw_if_not_writable()` and `dpctl::tensor::validation::AmpleMemory::throw_if_not_ample` like dpctl has?